metadata/src/main/java/org/apache/kafka/metadata/BrokerRegistration.java

```diff
@@ -213,12 +230,29 @@ public String toString() {
         bld.append("}");
         bld.append(", rack=").append(rack);
         bld.append(", fenced=").append(fenced);
+        bld.append(", inControlledShutdown=").append(inControlledShutdown);
         bld.append(")");
         return bld.toString();
     }
 
     public BrokerRegistration cloneWithFencing(boolean fencing) {
         return new BrokerRegistration(id, epoch, incarnationId, listeners,
-            supportedFeatures, rack, fencing);
+            supportedFeatures, rack, fencing, inControlledShutdown);
+    }
```

Comment:
What do you think about removing this method and replacing all of the calls with `cloneWith`.